public/js/app.js

```diff
@@ -0,0 +1,137 @@
+      } else if (secondsElapsed >= 2) {
+        timeElapsed = `${Math.floor(secondsElapsed)} seconds ago`;
+      } else if (secondsElapsed >= 1) {
+        timeElapsed = `${Math.floor(secondsElapsed)} second ago`;
+      }
+
+      // container div to hold each post;
+      let post = document.createElement('div');
+      post.className = 'postContainers';
+      contentArea.appendChild(post);
+
+      let postImage = document.createElement('img');
+      postImage.className = 'postImages';
+      postImage.src = dataImage;
+      postImage.onerror = function() {
+        postImage.src = defaultImg;
+      };
+      post.appendChild(postImage);
+
+      let postTitle = document.createElement('div');
+      postTitle.className = 'postTitles';
+      postTitle.innerHTML = dataTitle;
+      post.appendChild(postTitle);
+
+      let postDetail = document.createElement('div');
+      postDetail.className = 'postDetailContainers';
+      post.appendChild(postDetail);
+
+      let postAuthor = document.createElement('span');
+      postAuthor.className = 'postDetails';
+      postAuthor.innerHTML = dataAuthor;
+      postDetail.appendChild(postAuthor);
+
+      let postBullet1 = document.createElement('span');
+      postBullet1.className = 'postDetailBullets';
+      postBullet1.innerHTML = ' • ';
+      postDetail.appendChild(postBullet1);
+
+      let postCreated = document.createElement('span');
+      postCreated.className = 'postDetails';
+      postCreated.innerHTML = timeElapsed;
+      postDetail.appendChild(postCreated);
+
+      let postBullet2 = document.createElement('span');
+      postBullet2.className = 'postDetailBullets';
+      postBullet2.innerHTML = ' • ';
+      postDetail.appendChild(postBullet2);
+
+      let postLikes = document.createElement('span');
+      postLikes.className = 'postDetails';
+      postLikes.innerHTML = dataLikes + ' likes';
+      postDetail.appendChild(postLikes);
+
+      let postText = document.createElement('div');
+      postText.className = 'postTexts';
+      postText.innerHTML = dataText;
+      post.appendChild(postText);
+    }
+  }
+
+  const reddit = 'https://www.reddit.com/r/';
+  const redditReq = new XMLHttpRequest();
+  redditReq.addEventListener('load', redditListener);
+  redditReq.open('GET', reddit + threadName + '.json');
+  redditReq.send();
+}
+
+let menuClass = document.getElementsByClassName('menu');
+// dont include the final 'RANDOM' button in this event listener
+for (let i = 0; i < menuClass.length - 1; i++) {
+  menuClass[i].addEventListener('click', function() {
+    getRedditPage.apply(this, [this.dataset.thread]);
+  });
+}
+
+let randomPage = function() {
+  let pageArr = ['data', 'hawaii', 'food', 'hockey', 'travel', 'apples', 'laundry'];
+  let randomIndex = Math.floor(Math.random() * pageArr.length);
+  console.log(pageArr[randomIndex]);
+  return pageArr[randomIndex];
+};
+
+menu4.addEventListener('click', function() {
+  getRedditPage.apply(this, [randomPage()]);
```

Comment:
You don't need to use apply here either, `getRedditPage(randomPage())` would work just fine here.